add Crypto.md file decribes how to port and run crypto tests (#869)
ARMmbed/Handbook: docs/porting/psa/crypto.md

<h2 id="crypto-port">PSA CRYPTO</h2>

> For information about PSA-Crypto, please refer to [the PSA-Crypto repository](https://github.com/ARMmbed/mbedtls-psa).

Reviewer: Link to `https://github.com/ARMmbed/mbed-crypto` instead.

Also, use "Mbed Crypto" instead of "PSA Crypto" throughout this document.

## Testing
Mbed OS currently provide two test for PSA-Crypto
* Test for initialisation of the psa-crypto module
* Test for entropy injection feature

In order to run these tests make sure the following configuration is set in your target:

> 1.  "extra_labels" contain the label "PSA". For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1451) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7694) 

Reviewer: ```diff
-contain
+contains
```

> 2.  "MBEDTLS_PSA_CRYPTO_C" macro is enabled. For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1454) and [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7697)

Reviewer 2: Please replace 'Sequana' with 'Future Sequana', here and the whole doc.

Author: 👍

> 2.  "MBEDTLS_PSA_CRYPTO_C" macro is enabled. For example in [K64F](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L1454) and [Future Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7697)

Reviewer 3: @netanelgonen  should we go with exact label name as it is written in json file.
so it would be easier to lookup for the customer?
FUTURE_SEQUANA

Author: actually, the target name is FEATURE_SEQUANA_PSA, for the lookup the user can use the link. he doesn't need to look for it by himself... 
I don't care about changing it but, I think this is more readable.

> 3. "MBEDTLS_ENTROPY_NV_SEED" macro is enabled in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7673)

Reviewer: ```diff
-in wanted
+is wanted
```

> 4. "MBEDTLS_PLATFORM_NV_SEED_READ_MACRO" macro is set to mbed_default_seed_read" in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7674)

Reviewer: Unbalanced `"` here, around `mbed_default_seed_read`.

> 4. "MBEDTLS_PLATFORM_NV_SEED_READ_MACRO" macro is set to mbed_default_seed_read" in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7674)

Reviewer: ```diff
-in wanted
+is wanted
```

> 5. "MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO" macro is set to mbed_default_seed_write" in the SPE. If the device does not have TRNG or if entropy injection test in wanted. For example in [Sequana](https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L7674)

Reviewer: ```diff
-in wanted
+is wanted
```

### Compile and run

> in order to compile and run the PSA-Crypto Test run the following command:

Reviewer: ```diff
- in order
+ In order
```

<h2 id="crypto-port">Mbed Crypto</h2>

> Mbed OS provides Mbed Crypto for targets that require support for entropy injection. For information about Mbed Crypto, please refer to [the Mbed Crypto repository](https://github.com/ARMmbed/mbed-crypto).

Reviewer 3: it is more of "Mbed Crypto provides capability of entropy injection for target platforms"

Author: actually, I think that mbed-os is the right way to go, as the user takes mbed-os not mbed-crypto, and may not fully understand what you mean in mbed-crypto. please notice the document refer to tests in mbed-os and not mbed-crypto

> Mbed OS provides Mbed Crypto for targets that require support for entropy injection. For information about Mbed Crypto, please refer to [the Mbed Crypto repository](https://github.com/ARMmbed/mbed-crypto).

Reviewer 3: ```suggestion
Mbed Crypto library as part of Mbed OS provides support for entropy injection. For information about Mbed Crypto, please refer to [the Mbed Crypto repository](https://github.com/ARMmbed/mbed-crypto).
```

Author: the same comment as above, those tests are part of mbed-os and not mbed-crypto. I think mbed-os is correct.

Both targets newly ported to Mbed OS and existing targets that use entropy injection require passing tests to validate Mbed Crypto functionality.

> We run tests by default on some supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.

Reviewer 3: ```suggestion
We run tests by default on several supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.
```

> We run tests by default on some supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.

Reviewer 3: ```suggestion
We run tests by default on a list of supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.
```

Author: maybe keep the same as above, 'several'? it will be better to be consistent.

<h2 id="crypto-port">Mbed Crypto</h2>

Mbed OS provides Mbed Crypto for targets that require support for entropy injection. For information about Mbed Crypto, please refer to [the Mbed Crypto repository](https://github.com/ARMmbed/mbed-crypto).

Both targets newly ported to Mbed OS and existing targets that use entropy injection require passing tests to validate Mbed Crypto functionality.

We run tests by default on some supported boards. To run the tests on a custom board, you can also run the tests manually by specifying additional compiler flags from the command-line.

### Testing

Mbed OS currently provides two tests for Mbed Crypto:

- A test for initialization of the Mbed Crypto module.
- A test for entropy injection feature.

> To run these tests, make sure the following configuration is set in your target:

Reviewer 3: To run these tests, make sure your target configuration is set as follows:

Author: there is no difference between those two, do we need an additional commit for this?
